### ISSUE TITLE
In browser, mqtts:// gets replaced with ws:// rather than wss://

### ISSUE THREAD

**Issue by ksiegel** [mqttjs/MQTT.js#392]
When passing in a `mqtts://` URL in the browser, `connect/index.js` realizes that mqtts is unsupported and falls back to the first available protocol: `ws`

I'd like to see some logic that would replace mqtts with wss when mqtts is unavailable. 

I can submit a PR but would like some feedback first!

Code in question: 
https://github.com/mqttjs/MQTT.js/blob/master/lib/connect/index.js#L105

Something along the lines of:

``` js
  if (!protocols[opts.protocol]) {
    if(opts.protocol === 'mqtts' && 'function' === typeof protocols.wss){
      // mqtts:// becomes wss://
      opts.protocol = 'wss';
    } else {
      // otherwise default to trying first available protocol
      opts.protocol = protocolList.filter(function (key) {
        return 'function' === typeof protocols[key];
      })[0];
    }
  }
```

Thanks!

**Comment by mcollina** [comment]
Yes, please send a PR! However you might want to overhaul that logic completely. The whole protocolList thing was used to to avoid having those kind of ifs, and it fails the purpose. Maybe a simple regexp if we are in a browser?

**Comment by mcollina** [comment]
Closing for no activity.

**Comment by mbrandau** [comment]
Seems like this problem was fixed in #518. Is there a reason why this issue is still open? @mcollina @ksiegel
